state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2414,10 +2412,9 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				return Err(DepositFailedReason::BelowMinimumDeposit);
 			}
 			if T::AllowTransactionReports::get() {
-				if let (Some(tx_ids), Some(broker_id)) = (
-					deposit_details.deposit_ids(),
-					origin.broker_id().map(|id| Self::is_whitelisted_broker_or(id.clone())),
-				) {
+				if let (Some(tx_ids), Some(broker_id)) =
+					(deposit_details.deposit_ids(), origin.broker_id())
+				{
```

Comment:
The only thing that stands out to me is that we don't run this code if `origin.broker_id() == None`, even though the ScreeningBrokerId might have requested rejection.

Though the behaviour before this commit was the same, so I assume that's correct?

Reply:
Yeah - now that we enforce a broker, this will always return Some. It's not yet explicit in the storage types, however. This will follow in the 1.9 release.

Reply:
(See [here](https://github.com/chainflip-io/chainflip-backend/pull/5630#issuecomment-2696950439))